src/ui/projects/CreateScenePage.js

```diff
@@ -0,0 +1,366 @@
+        setSceneInfo({
+          name: scene.name || "",
+          creatorAttribution: (scene.attributions && scene.attributions.creator) || "",
+          allowRemixing: scene.allow_remixing,
+          allowPromotion: scene.allow_promotion
+        });
+        setThumbnailUrl(scene.screenshot_url);
+        setSceneId(scene.scene_id);
+        setSceneUrl(scene.url);
+        setIsLoading(false);
+      }
+      doInitialLoad();
+    },
+    [match, api, setSceneInfo, setThumbnailUrl, setSceneId, setSceneUrl, setIsLoading]
+  );
+
+  const onChangeName = useCallback(
+    function(name) {
+      setSceneInfo({ ...sceneInfo, name });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeCreatorAttribution = useCallback(
+    function(creatorAttribution) {
+      setSceneInfo({ ...sceneInfo, creatorAttribution });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeAllowRemixing = useCallback(
+    function(allowRemixing) {
+      setSceneInfo({ ...sceneInfo, allowRemixing });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeAllowPromotion = useCallback(
+    function(allowPromotion) {
+      setSceneInfo({ ...sceneInfo, allowPromotion });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeGlbFile = useCallback(
+    function([file]) {
+      setGlbFile(file);
+    },
+    [setGlbFile]
+  );
+
+  const onChangeThumbnailFile = useCallback(
+    function([file]) {
+      if (thumbnailUrl && thumbnailUrl.indexOf("data:") === 0) {
+        URL.revokeObjectURL(thumbnailUrl);
+      }
+
+      setThumbnailFile(file);
+
+      // For preview
+      const reader = new FileReader();
+      reader.onload = e => {
+        setThumbnailUrl(e.target.result);
+      };
+      reader.readAsDataURL(file);
+    },
+    [thumbnailUrl, setThumbnailUrl, setThumbnailFile]
+  );
+
+  const onPublish = useCallback(
+    function(e) {
+      e.preventDefault();
+
+      setError(null);
+      setIsUploading(true);
+
+      const abortController = new AbortController();
+
+      api
+        .publishGLBScene(
+          thumbnailFile,
+          glbFile,
+          {
+            name: sceneInfo.name,
+            allow_remixing: sceneInfo.allowRemixing,
+            allow_promotion: sceneInfo.allowPromotion,
+            attributions: {
+              creator: sceneInfo.creatorAttribution,
+              content: []
+            }
+          },
+          abortController.signal,
+          sceneId
+        )
+        .then(() => history.push("/projects"))
+        .catch(e => {
+          setIsUploading(false);
+          setError(e.message);
+        });
+    },
+    [thumbnailFile, glbFile, sceneInfo, sceneId, setIsUploading, api, history, setError]
+  );
+
+  const isNew = !sceneId;
+
+  const content = isLoading ? (
+    <ProgressBar />
+  ) : (
+    <>
+      <SceneUploadHeader>
+        <h1>{isNew ? "Publish Scene From GLB" : "Update GLB Scene"}</h1>
```

Comment:
Again, we should revisit this copy. I think this is clearer than the `Upload GLB` button though. Maybe we need some descriptive copy to go along with this and tell the user what is happening?

Reply:
As per discussion keeping it this way on this button.